Remove explicit "extends @Nullable Object."

cpovirk opened the pull request:
It is supposed to be a no-op. As such, the stub files usually omit it.

Currently, it actually changes behavior in a bad way:
https://github.com/typetools/checker-framework/issues/2995

Until that bug is fixed, this commit serves as a workaround.

the1derer commented on make/common/SetupJavaCompilers.gmk:
```diff
@@ -35,17 +35,21 @@ DISABLE_WARNINGS ?= -Xlint:all,-deprecation,-removal,-unchecked,-rawtypes,-cast,
 # The BOOT_JAVAC setup uses the boot jdk compiler to compile the tools
 # and the interim javac, to be run by the boot jdk.
+# (Checker Framework Comment: Temporarily removed -Werror.)
+#       -Xlint:all$(COMMA)-deprecation$(COMMA)-options -Werror, \
```
@mernst
I think this will give errors as special characters have been not escaped. I have corrected this in #25. I think we should close this PR in favor of #25

mernst replied:
Sorry, I don't understand your comment.
What errors will this give, and why?

the1derer replied:
On running `make images` on `mernst:add-checker.jar-to-build`:
```log
Building target 'images' in configuration 'linux-x86_64-normal-server-release'
ToolsLangtools.gmk:37: *** The Java compilation BUILD_TOOLS_LANGTOOLS refers to a non-existant java compiler setup BOOT_JAVAC.  Stop.
make[2]: *** [make/Main.gmk:73: buildtools-langtools] Error 2
make[2]: *** Waiting for unfinished jobs....

ERROR: Build failed for target 'images' in configuration 'linux-x86_64-normal-server-release' (exit code 2) 

No indication of failed target found.
Hint: Try searching the build log for '] Error'.
Hint: See doc/building.html#troubleshooting for assistance.

make[1]: *** [/mnt/Work/Checker_Framework/jdk/make/Init.gmk:305: main] Error 2
make: *** [/mnt/Work/Checker_Framework/jdk/make/Init.gmk:186: images] Error 2
```
This is because in the` #       -Xlint:all$(COMMA)-deprecation$(COMMA)-options -Werror, \` we have to escape `$`, `(`  `)` and `\`

cpovirk committed:
Remove explicit "extends @Nullable Object."

It is supposed to be a no-op. As such, the stub files usually omit it.

Currently, it actually changes behavior in a bad way:
https://github.com/typetools/checker-framework/issues/2995

Until that bug is fixed, this commit serves as a workaround.
